presidio-analyzer/presidio_analyzer/recognizer_registry_provider.py

```diff
@@ -0,0 +1,156 @@
+        return [{"supported_language": language["language"], "context": language["context"]} for language in recognizer["supported_languages"]]
+
+    def _split_recognizers(self, recognizers: Union[Dict[str, Any], str]) -> Tuple[List[Dict[str, Any]],List[Dict[str, Any]]]:
+        """
+        splits the recognizer list to predefined and custom. All recognizers are custom by default though type: 'custom' can be mentioned as well. 
+        This function supports the previous format as well.
+        """
+        predefined =[]
+        custom =[]
+
+        predefined = [recognizer for recognizer in recognizers if isinstance(recognizer, str) or ("type" in recognizer and recognizer["type"] == "predefined")]
+        custom = [recognizer for recognizer in recognizers if not isinstance(recognizer, str) and ("type" not in recognizer or recognizer["type"] == "custom")]
+        return predefined, custom
+
+    def _create_custom_recognizers(self, recognizer: Dict) -> List[EntityRecognizer]:
+        # legacy recognizer
+        if "supported_language" in recognizer:
+            return [PatternRecognizer.from_dict(recognizer)]
+
+        recognizers = []
+
+        for supported_language in self._get_recognizer_languages(recognizer):
+            copied_recognizer = {k: v for k, v in recognizer.items() if k not in ["enabled", "type", "supported_languages"]}
+            recognizers.append(PatternRecognizer.from_dict({**copied_recognizer, **supported_language}))
+
+        return recognizers
+    
+    default_values = {
+        "supported_languages": ["en"],
+        "recognizers": [],
+        "global_regex_flags": re.DOTALL | re.MULTILINE | re.IGNORECASE
+    }
+
+    def create_recognizer_registry(self) -> RecognizerRegistry:
+        fields = {
+            "supported_languages": None,
+            "recognizers": None,
+            "global_regex_flags": None
+        }
+
+        for field in fields.keys():
+            if field not in self.configuration:
+                logger.warning(f'{field} not present in configuration, using default value instead: {self.default_values[field]}')
+            fields[field] = self.configuration.get(field, self.default_values[field])
+
+        self.supported_languages = fields["supported_languages"]
+
+        recognizer_instances = []
+        predefined, custom = self._split_recognizers(fields["recognizers"])
+        for recognizer in predefined:
+            for language in self._get_recognizer_languages(recognizer):
+                if self._is_recognizer_enabled(recognizer):
+                    copied_recognizer = {k: v for k, v in self._get_recognizer_items(recognizer) if k not in ["enabled", "type", "supported_languages", "name"]}
+                    recognizer_instances.append(getattr(presidio_analyzer.predefined_recognizers, self._get_recognizer_name(recognizer), None)(**{**copied_recognizer, **language}))
```

Comment:
WDYT about fetching all subclasses (recursively) instead of `presidio_analyzer.predefined_recognizers`? This could potentially allow more flexibility for users having their own predefined recognizers in code, but not part of the package (as they didn't clone the repo, just used the package).

Reply:
I would suggest to simplify this line: `(**{**copied_recognizer, **language}))`. Perhaps create the unified dictionary prior to calling the init.